File: app/src/main/java/org/dhis2/AppModule.java
package org.dhis2;

import android.content.Context;

import androidx.annotation.NonNull;

import org.dhis2.commons.resources.ResourceManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private final App application;

    public AppModule(@NonNull App application) {
        this.application = application;
    }

    @Provides
    @Singleton
    Context context() {
        return application;
    }

    @Provides
    @Singleton
    ResourceManager resources() {
        return new ResourceManager(application);
    }
}

File: app/src/main/java/org/dhis2/AppModule.kt
package org.dhis2

import android.content.Context
import dagger.Module
import dagger.Provides
import org.dhis2.commons.resources.ColorUtils
import org.dhis2.data.service.CheckVersionWorker
import org.dhis2.data.service.SyncGranularWorker
import org.dhis2.usescases.main.mainModule
import org.koin.androidx.workmanager.dsl.workerOf
import org.koin.dsl.module
import javax.inject.Singleton

@Module
class AppModule(
    private val application: App,
) {
    @Provides
    @Singleton
    fun context(): Context = application

    @Provides
    @Singleton
    fun colorUtils(): ColorUtils = ColorUtils()
}

val appModule =
    module {
        includes(mainModule)
        workerOf(::SyncGranularWorker)
        workerOf(::CheckVersionWorker)
    }
